vernac/record.ml

```diff
@@ -771,7 +771,27 @@ let interp_structure udecl kind ~template ~cumulative ~poly ~primitive_proj fini
     pre_process_structure udecl kind ~poly records in
   interp_structure_core ~cumulative finite ~univs ~variances ~primitive_proj impargs params template ~projections_kind data
 
-let declare_structure { Record_decl.mie; primitive_proj; impls; globnames; global_univ_decls; projunivs; ubinders; projections_kind; poly; records } =
+let autocut ind projs =
+  let open Hints in
+  let rec fold f = function
+    | [] -> assert false
+    | [x] -> x
+    | x :: rest -> f x (fold f rest)
+  in
+  let seql = fold (fun x y -> PathSeq (x,y)) in
+  let orl = fold (fun x y -> PathOr (x,y)) in
+  let starany = PathStar (PathAtom PathAny) in
+  let atom gr = PathAtom (PathHints [gr]) in
+  let projs = List.filter_map (fun p ->
+      Option.map (fun c -> atom (GlobRef.ConstRef c)) p.Structure.proj_body)
+      projs
+  in
+  if List.is_empty projs then ()
+  else
+    let entry = HintsCutEntry (seql [starany; orl projs; starany; atom (GlobRef.ConstructRef (ind,1))]) in
```

Comment:
Does the bench approve if we do
```suggestion
    let entry = HintsCutEntry (seql [starany; orl projs; atom (GlobRef.ConstructRef (ind,1))]) in
```
which should be good enough?

Reply:
let's see